pkg/redpanda/client.go

```diff
@@ -0,0 +1,210 @@
+package redpanda
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net"
+	"slices"
+	"strings"
+
+	"github.com/redpanda-data/helm-charts/charts/redpanda"
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl"
+	"github.com/twmb/franz-go/pkg/sasl/plain"
+	"github.com/twmb/franz-go/pkg/sasl/scram"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	ErrServerCertificateNotFound          = errors.New("kafka TLS certificate not found")
+	ErrServerCertificatePublicKeyNotFound = errors.New("kafka TLS certificate does not contain a public key")
+
+	ErrClientCertificateNotFound           = errors.New("client TLS certificate not found")
+	ErrClientCertificatePublicKeyNotFound  = errors.New("client TLS certificate does not contain a public key")
+	ErrClientCertificatePrivateKeyNotFound = errors.New("client TLS certificate does not contain a private key")
+
+	ErrSASLSecretNotFound          = errors.New("users secret not found")
+	ErrSASLSecretKeyNotFound       = errors.New("users secret key not found")
+	ErrSASLSecretSuperuserNotFound = errors.New("users secret has no users")
+
+	supportedSASLMechanisms = []string{
+		"PLAIN", "SCRAM-SHA-256", "SCRAM-SHA-512",
+	}
+)
+
+func firstUser(data []byte) (user string, password string, mechanism string) {
```

Comment:
nit: prefer to place unexpected helpers at the bottom of the file and exported functions at the top in some degree of ordering.